pwa/src/components/common/layout.tsx

```diff
@@ -3,53 +3,59 @@ import Footer from './../footer/footer'
 import MainMenu from "./menu";
 import { Helmet } from "react-helmet";
 import "bootstrap/dist/css/bootstrap.css";
-import Header from "./header";
 import { APIProvider } from "../../apiService/apiContext";
 import APIService from "../../apiService/apiService";
 import { isLoggedIn } from "../../services/auth";
 import Login from "../../pages/login";
+import { AlertProvider, AlertProps } from "../../context/alertContext";
 import WelcomeModal from "../welcomeModal/welcomeModal";
 import favicon from "../../images/conduction_logo_blauw.svg";
-
+import Alert from "../alert/alert";
+import {HeaderProps, HeaderProvider} from "../../context/headerContext";
+import Header from "./header";
 /**
  * This components renders a layout which is renders the menu, footer and container surrounding main body of pages.
  *
  * @param {object} children Content that is rendered as body.
- * @param {string|null} title Title for the page.
- * @param {string|null} subText Subtext for the site.
- * @returns TSX of the generated Layout.
+ * @returns JSX of the generated Layout.
  */
-export default function Layout({ children, title = "", subtext = "" }) {
+export default function Layout({ children}) {
   const [API, setAPI] = React.useState<APIService>(null)
+  const [alert, setAlert] = React.useState<AlertProps>(null);
+  const [header, setHeader] = React.useState<HeaderProps>(null);
 
   React.useEffect(() => {
     !API && setAPI(new APIService(sessionStorage.getItem('jwt')))
   }, [API])
 
   return (
     API &&
-      <APIProvider value={API}>
+    <APIProvider value={API}>
+      <AlertProvider value={[alert, setAlert]}>
+        <HeaderProvider value={[header, setHeader]}>
```

Comment:
We're chaining multiple ContextProviders here. Which is still manageable here because it's only three. However, when we need more and more in our Context, this won't work anymore.

Therefore, we will need to create a more generic place where we can chain these ContextProviders and then within the layout we only have to wrap it like (e.g.) `<ContextProvider value={context}>`.

Could you create a JIRA issue for this so we can do this in the future?